solver/src/driver.rs

```diff
@@ -97,16 +134,15 @@ impl Driver {
                 info!(
                     "{} found solution with objective value: {}",
                     solver,
-                    settlement.objective_value(/* estimated_prices*/)
+                    settlement.objective_value(&estimated_prices)
                 );
                 Some((solver, settlement))
             })
             .collect();
 
         // Sort by key in descending order
-        settlements.sort_by_key(|(_, settlement)| {
-            Reverse(settlement.objective_value(/* estimated_prices*/))
-        });
+        settlements
+            .sort_by_key(|(_, settlement)| Reverse(settlement.objective_value(&estimated_prices)));
```

Comment:
Done in fd1b65. Perhaps an even better solution would be to store the objective value in the settlement object upon construction, since otherwise we still compute it twice .